src/sas/qtgui/Calculators/GenericScatteringCalculator.py

```diff
@@ -857,10 +864,70 @@ def update_gui(self):
             self.txtZstepsize.setText(GuiUtils.formatValue(self.mag_sld_data.zstepsize))
         # otherwise leave as set since editable by user
 
+        # update the value of the Radius of Gyration with values from the loaded data
+        if self.is_nuc:
+            if self.nuc_sld_data.is_elements:
+                self.txtROG.setText(str("N/A for Elements"))
+            else:
+                self.txtROG.setText(str(round(self.radius_of_gyration(),1)) + " Å")
+        elif self.is_mag:
+            self.txtROG.setText(str("N/A for magnetic data"))
+        else:
+            self.txtROG.setText(str("N/A for no data"))
+            
+
         # If nodes or stepsize changed then this may effect what values are allowed
         self.gui_text_changed(sender=self.txtNoQBins)
         self.gui_text_changed(sender=self.txtQxMax)
     
+    def radius_of_gyration(self):
+        #Calculate Center of Mass(CoM) First
+        CoM = self.centerOfMass()
+
+        #Now Calculate RoG
+        RoGNumerator = RoGDenominator = 0.0
+
+        for i in range(len(self.nuc_sld_data.pos_x)):
+            coordinates = [0.0,0.0,0.0]
+            coordinates = [float(self.nuc_sld_data.pos_x[i]),float(self.nuc_sld_data.pos_y[i]),float(self.nuc_sld_data.pos_z[i])]
```

Comment:
Since you're explicitly creating the list in the next line, line 891 is not needed.